### PR TITLE
Fixes #431 - Add http-proxy support

### LINKED ISSUE #431: Add support for http-proxy component.

### PR DESCRIPTION
**Pull request by jameerpathan111**
Fixes #431

### CODE REVIEW

File: airgun/entities/http_proxy.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    def read(self, entity_name, widget_names=None):
+        """Return dict with properties of http-proxy."""
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+        return view.read(widget_names=widget_names)
+
+    def update(self, entity_name, values):
+        """Update an existing http-proxy."""
+        view = self.navigate_to(self, 'Edit', entity_name=entity_name)
+
```
Comment (omkarkhatavkar) [inline airgun/entities/http_proxy.py]: extra space here ?

Comment (jameerpathan111) [reply airgun/entities/http_proxy.py]: Thanks, updated PR.

File: airgun/views/http_proxy.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+
+class HTTPProxyView(BaseLoggedInView, SearchableViewMixin):
+    title = Text("//h1[text()='HTTP Proxies']")
+    new = Text("//a[contains(@href, 'http_proxies/new')]")
+    table = SatTable(
+        './/table',
+        column_widgets={
+            'Name': Text('./a'),
+            'URL': Text('./a'),
```
Comment (omkarkhatavkar) [inline airgun/views/http_proxy.py]: './a' is this valid? Do both have same locator?

File: airgun/views/http_proxy.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+from airgun.widgets import (
+    MultiSelect,
+    SatTable,
+)
+
+
+class HTTPProxyView(BaseLoggedInView, SearchableViewMixin):
+    title = Text("//h1[text()='HTTP Proxies']")
+    new = Text("//a[contains(@href, 'http_proxies/new')]")
+    table = SatTable(
```
Comment (omkarkhatavkar) [inline airgun/views/http_proxy.py]: use Table instead SatTable

File: airgun/entities/http_proxy.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        view.validations.assert_no_errors()
+        view.flash.assert_no_error()
+        view.flash.dismiss()
+
+    def delete(self, entity_name):
+        """Delete existing http-proxy entity"""
+        view = self.navigate_to(self, 'All')
+        self.search(entity_name)
+        view.table.row(Name=entity_name)['Actions'].widget.click(
+            handle_alert=True)
```
Comment (omkarkhatavkar) [inline airgun/entities/http_proxy.py]: where are you deleting here?

Comment (jameerpathan111) [reply airgun/entities/http_proxy.py]: There is only one button in Action column i.e "Delete". So above code works.
I can also use `view.table.row(name=entity_name)['Actions'].widget.fill('Delete')`

Comment (jameerpathan111) [reply airgun/entities/http_proxy.py]: There is only one button in Action column i.e. "Delete".

File: airgun/views/http_proxy.py (hunk 3 above)
Comment (jameerpathan111) [reply airgun/views/http_proxy.py]: updated PR.

File: airgun/views/http_proxy.py (hunk 2 above)
Comment (jameerpathan111) [reply airgun/views/http_proxy.py]: yes, it's valid.

**Comment by jameerpathan111** [comment]
Test result: 
[https://github.com/SatelliteQE/robottelo/pull/7530#issuecomment-563891203](https://github.com/SatelliteQE/robottelo/pull/7530#issuecomment-563891203)

File: airgun/session.py [hunk 5]
```diff
@@ -404,6 +405,11 @@ def hostgroup(self):
         """Instance of Host Group entity."""
         return self._open(HostGroupEntity)
 
+    @cached_property
+    def http_proxy(self):
+        """Instance of Host Group entity."""
```
Comment (lhellebr) [inline airgun/session.py]: Host Group -> HTTP Proxy

Comment (jameerpathan111) [reply airgun/session.py]: Thanks :)
Updated PR.

**Pull request merged**